src/QuadVelocityController.cpp

```diff
@@ -88,48 +104,50 @@ bool QuadVelocityController::update(const ros::Time& time,
     double throttle_output;
     double pitch_output;
     double roll_output;
-    double yaw_output;
 
     // Update throttle PID loop
-    success = throttle_pid_.update(velocity.linear.z, time, throttle_output);
+    success = throttle_pid_.update(velocity.z, time, throttle_output);
     if (!success) {
         ROS_WARN("Throttle PID update failed in QuadVelocityController::update");
         return false;
     }
 
     // Update pitch PID loop
-    success = pitch_pid_.update(velocity.linear.x, time, pitch_output);
+    success = pitch_pid_.update(std::cos(current_yaw) * velocity.x
```

Comment:
Would it improve readability to put the transformation code onto the line above with a descriptive variable name saying what frame the vector was transformed to?